src/rgw/rgw_sal_motr.cc

```diff
@@ -2346,18 +2356,23 @@ int MotrObject::update_version_entries(const DoutPrefixProvider *dpp)
     ent.flags = rgw_bucket_dir_entry::FLAG_VER;
     string key;
     if (ent.key.instance.empty())
+    { 
       key = ent.key.name;
+    }
     else {
-      char buf[ent.key.name.size() + ent.key.instance.size() + 16];
-      snprintf(buf, sizeof(buf), "%s[%s]", ent.key.name.c_str(), ent.key.instance.c_str());
-      key = buf;
+      if (ent.key.instance == "null")
+        key = keys[i];
+      else{
+          char buf[ent.key.name.size() + ent.key.instance.size() + 16];
```

Comment:
whats this buffer for, and what 16 signifies here? If its going to get reused somewhere, use it as macro

Reply:
Removed this code as it is no longer required here.